crates/sp-domains/src/fraud_proof.rs

```diff
@@ -179,20 +180,56 @@ pub enum VerificationError {
         error("Oneshot error when verifying fraud proof in tx pool: {0}")
     )]
     Oneshot(String),
+    #[cfg_attr(feature = "thiserror", error("Tx is in range: {extrinsic_index}"))]
+    TxIsInRange { extrinsic_index: u32 },
 }
 
-// TODO: Define rest of the fraud proof fields
 #[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
-pub struct MissingInvalidBundleEntryFraudProof {
-    domain_id: DomainId,
-    bundle_index: u32,
+pub enum MissingBundleAdditionalData {
+    OutOfRangeTx { extrinsic_index: u32 },
 }
 
-impl MissingInvalidBundleEntryFraudProof {
-    pub fn new(domain_id: DomainId, bundle_index: u32) -> Self {
+#[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone)]
+pub struct MissingInvalidBundleEntryFraudProof<Number, Hash, DomainNumber, DomainHash> {
+    pub domain_id: DomainId,
+    pub bad_receipt_hash: H256,
+    pub consensus_block_hash: Hash,
+    pub parent_domain_block_hash: DomainHash,
+    pub bundle_index: u32,
+    pub opaque_bundle_with_proof:
+        OpaqueBundleWithProof<Number, Hash, DomainNumber, DomainHash, Balance>,
```

Comment:
With #1995 we don't need to include the whole bundle in the fraud proof, plz take a look at the PR description.